src/bit-fields-macros/src/lib.rs

```diff
@@ -0,0 +1,1070 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Do not use this. Use [bit-fields](https://docs.rs/bit-fields).
```

Comment:
`bit-fields` and `bit-fields-macros` are intended to be used as a single crate. Attempting to use `bit-fields-macros` on its own, will simply errors, this documentation is here to emphasize this. The documentation relating to why this crate is different is under `firecracker/src/bit-fields/src/lib.rs`

> Also maybe it is not a bad idea to find a custom name for it, maybe cpuid-compiler (the same way we have seccompiler)?

I have named it to describe the functionality it offers and what it does rather than how we use it. I find this clearest. Given this is a common component we may see other uses for and we may publish to crates.io in future.

Reply:
I have removed the link, replacing it with `//! Do not use this. Use `bit-fields` instead.`.